include/nil/actor/zk/snark/arithmetization/plonk/constraint.hpp

```diff
@@ -138,27 +134,21 @@ namespace nil {
                                     assignment =
                                         math::polynomial_shift(assignment, domain->get_domain_element(var.rotation));
                                 }
-
-                                term_value = term_value * assignment;
-                            }
-                            acc = acc + term_value;
-                        }
-                        return make_ready_future<math::polynomial<typename VariableType::assignment_type>>(acc);
+                                return assignment;
+                            });
+                        return evaluator.evaluate().get();
                     }
 
                     template<typename ArithmetizationParams>
-                    future<math::polynomial_dfs<typename VariableType::assignment_type>>
+                    math::polynomial_dfs<typename VariableType::assignment_type>

```

Comment:
And it's strange that we have two function `evaluate` but one of them return future, another - polynomial_dfs. But they both parallel.